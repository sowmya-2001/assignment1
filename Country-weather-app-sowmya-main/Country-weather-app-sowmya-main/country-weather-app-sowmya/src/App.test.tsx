import { render, screen, fireEvent, act } from "@testing-library/react";

describe("test Input Field component", () => {
  test("render Input Field component", () => {
    const InputFieldProps = {
      onChange: jest.fn(),
      type: "text",
      id: "header-search",
      placeholder: "Country Name",
    };
    render(<input {...InputFieldProps} />);
    screen.debug();
  });

  test("snapshots testing", () => {
    const InputProps = {
      onChange: jest.fn(),
      className: "inputField",
      placeHolder: "Country Name",
    };
    const { container } = render(<input {...InputProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("check if on change is called", () => {
    const InputProps = {
      onChange: jest.fn(),
      type: "text",
      id: "header-search",
      placeholder: "Country Name",
    };
    render(<input {...InputProps} />);
    const inputElement = screen.getByDisplayValue("");
    screen.debug();
    fireEvent.change(inputElement, { target: { value: "France" } });
    expect(InputProps.onChange).toHaveBeenCalledTimes(1);
  });
});
