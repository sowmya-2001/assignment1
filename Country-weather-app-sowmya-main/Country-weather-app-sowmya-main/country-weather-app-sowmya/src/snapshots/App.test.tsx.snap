// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input Field component snapshot testing 1`] = `
<input
  class="inputField"
  placeholder="Country Name"
/>
`;
